app/controllers/articles_controller.rb

```diff
@@ -14,26 +14,26 @@ def new
   end
 
   def create
-    @article = Article.new
     user = User.find_by_username(article_params[:user][:username])
+    @article = Article.new
 
     @article.title = article_params[:title]
     @article.description = article_params[:description]
     @article.user = user
-    
+
     if @article.save
       flash[:success] = 'Article was successfully created'
       redirect_to article_path(@article)
     else
       render 'new'
     end
   end
-  
+
   def update
     user = User.find_by(username: article_params[:user][:username])
-    
-    if @article.update({title: article_params[:title], description: article_params[:description], user: user})
-      flash[:success] = "Article was updated"
+
+    if @article.update(title: article_params[:title], description: article_params[:description], user: user)
```

Comment:
You can write it just
```suggestion
    if @article.update(article_params)
```